pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -98,8 +98,8 @@ func (alloc *Action) Execute(ssn *framework.Session) {
 	allNodes := ssn.NodeList
 	predicateFn := func(task *api.TaskInfo, node *api.NodeInfo) error {
 		// Check for Resource Predicate
-		if !task.InitResreq.LessEqual(node.FutureIdle(), api.Zero) {
-			return api.NewFitError(task, node, api.NodeResourceFitFailed)
+		if ok, reason := task.InitResreq.CheckResource(node.FutureIdle(), api.Zero); !ok {
```

Comment:
@gj199575 LessEqual is an imporment foundamental function. What's the difference between the CheckResource and LessEqual, and why we have to add this new function, Would you provide detail reason?

Reply:
yes, LessEqual is used by many place, And the new function CheckResource   is  ready to slove the problem  in  https://github.com/volcano-sh/volcano/issues/2774.  But it is the same as LessEqual  in code。
And after I solve  https://github.com/volcano-sh/volcano/issues/2774 。 
I will continue change to  merge LessEqual and CheckResource to a function 。

Reply:
> @gj199575 LessEqual is an imporment foundamental function. What's the difference between the CheckResource and LessEqual, and why we have to add this new function, Would you provide detail reason?

And it is a huge project if I change LessEqual function .So I supply a new function LessEqualWithReason 。 And I will  merge LEssEqual and LessEqualWithReason  to a funtion 。